syntax = "proto3";

package logs;

//Log Retrieving Service Definition
service LogRetriever {
    rpc GetMessages (GetMessageReq) returns (MessagesReply) {}
}

//The message formate for a request, includes a timstamp and a delta interval
message GetMessageReq {
    string date = 1;
    string start = 2;
    string delta = 3;
}

//The message reply format
message MessagesReply {
    string message = 1;
    repeated string coded_logs = 2;
}